lib/graphql/batch/execution_strategy.rb

```diff
@@ -49,7 +49,13 @@ def each_promise_in_entry(obj, key, value, &block)
     class FieldResolution < GraphQL::Query::SerialExecution::FieldResolution
       def get_finished_value(raw_value)
         if raw_value.is_a?(::Promise)
-          raw_value.then { |result| super(result) }
+          on_reject = lambda do |error|
+            raise error unless error.is_a?(GraphQL::ExecutionError)
+            error.ast_node = ast_node
+            query.context.errors << error
```

Comment:
Perhaps this could be done in the upstream get_finished_value if a GraphQL::ExecutionError instance is passed into it instead of doing this in the result method upstream.